interpreter/src/functions.rs

```diff
@@ -697,10 +696,13 @@ mod tests {
     #[test]
     fn test_has() {
         let tests = vec![
-            ("map has", "has(foo.bar) == true"),
             ("map has", "has(foo.bar) == true"),
             ("map not has", "has(foo.baz) == false"),
-            ("map deep not has", "has(foo.baz.bar) == false"),
+            // ("map deep not has", "has(foo.baz.bar) == false"), // todo this isn't correct is it?
```

Comment:
This is indeed not correct, and should [fail with `failed to evaluate: no such key: baz`](https://playcel.undistro.io/?content=H4sIAAAAAAAAA3WSTWvDMAyG%2F4qWHdpCl7Jrb2UEtpFtZWOUQS5qoiYG2wq204%2BN%2FffJXqD0UIxBSNbzvrL9k9Wks2W2WMCGdM2GIDCEjuChKGGt8dQ6HmxzU1k5cpkD5QEtKBvIYR3UnoSxXXlPZqtP0POBHDVAdq8cW0M2RDYde82OpLOJMTmVKgcVun9dNoYtFMfekfdKwhJtO2BLMBX9WR6dJDd3sHEqEJx4cMkanXs6kR4PfXpKYHSEIKUYe9UQ7NiJ%2BX4I0GDAucTwtXopQdLPH2%2BvsW4wjJR1JMPkfbCTNMUe9YBXxLFFZX1ISmeBEVSM88e%2BmrUmuTnp4Z0Q0PSa%2FGjM98phrFWyOvTTHXO%2BxW%2FZbgaVzeZZxD5FAXnBW3iUmeObXMpeGSuPVCEuKwsgxCXcJ6ThhoQWf8XvHw7fupMdAgAA)